kubernetes/samples/scripts/create-fmw-infrastructure-domain/domain-home-in-image/create-domain.sh

```diff
@@ -173,60 +214,85 @@ function getDockerSample {
-    passwordFile="${dockerPropsDir}/adminpass.properties"
-  fi
-  
-  sed -i -e "s|myusername|${username}|g" $usernameFile
-  sed -i -e "s|welcome1|${password}|g" $passwordFile
-
-  sed -i -e "s|INFRA08|${rcuSchemaPrefix}|g" $rcuPropFile
-  sed -i -e "s|InfraDB:1521/InfraPDB1|${rcuDatabaseURL}|g" $rcuPropFile
 
-  cp -f ${scriptDir}/common/Dockerfile ${dockerDir}/Dockerfile
-  cp -f ${scriptDir}/common/createFMWDomain.sh ${dockerDir}/container-scripts
+  echo @@ "Info: WIT_DIR is ${WIT_DIR}"
+
+  domainPropertiesOutput="${domainOutputDir}/domain.properties"
+
+  echo @@ "Info: Invoking WebLogic Image Tool to create a WebLogic domain at '${domainHome}' from image '${domainHomeImageBase}' and tagging the resulting image as '${BUILD_IMAGE_TAG}'."
+
+  if [ "${mode}" == "wlst" ]; then
+      additionalBuildCommandsOutput="${domainOutputDir}/additional-build-commands"
+      additionalBuildCommandsTemplate="wlst/additional-build-commands-template"
+
+      # Generate the additional-build-commands file that will be used when creating the weblogic domain
+      echo @@ "Info: Generating ${additionalBuildCommandsOutput} from ${additionalBuildCommandsTemplate}"
+
+      cp ${scriptDir}/${additionalBuildCommandsTemplate} ${additionalBuildCommandsOutput} || exit 1
+      sed -i -e "s:%DOMAIN_HOME%:${domainHome}:g" ${additionalBuildCommandsOutput}
+
+      createDomainWlstScriptCopy="${domainOutputDir}/createFMWDomain.py"
+      cp ${scriptDir}/${createDomainWlstScript} ${createDomainWlstScriptCopy} || exit 1
+
+      cmd="
+        $WIT_DIR/imagetool/bin/imagetool.sh update
+          --fromImage \"$domainHomeImageBase\"
+          --tag \"${BUILD_IMAGE_TAG}\"
+          --wdtOperation CREATE
+          --wdtVersion ${WDT_VERSION}
+          --wdtDomainHome \"${domainHome}\"
+          --additionalBuildCommands ${additionalBuildCommandsOutput}
+          --additionalBuildFiles \"${scriptDir}/wlst/createFMWDomain.sh,${createDomainWlstScriptCopy},${domainPropertiesOutput}\"
+          --chown=oracle:root
+        "
+  else # wdt
+    createDomainWdtModelCopy="${domainOutputDir}/wdt_model.yaml"
+    cp ${scriptDir}/${createDomainWdtModel} ${createDomainWdtModelCopy} || exit 1
+
+    if [ -n "${wdtEncryptKey}" ]; then
+      echo @@ "Info: An encryption key is provided, encrypting passwords in WDT properties file"
+      wdtEncryptionKeyFile=${domainOutputDir}/wdt_encrypt_key
+      echo  -e "${wdtEncryptKey}" > "${wdtEncryptionKeyFile}"
+      domainOutputDirFullPath="$( cd "$( dirname "${domainPropertiesOutput}" )" && pwd)"
+      encrypt_model ${domainOutputDirFullPath} wdt_model.yaml wdt_encrypt_key domain.properties || exit 1
+    fi
 
-  if [ ! -z $domainHomeImageBase ]; then
-    sed -i -e "s|\(FROM \).*|\1 ${domainHomeImageBase}|g" ${dockerDir}/Dockerfile
+    echo @@ "Info: dumping output of ${domainPropertiesOutput}"
+    sed 's/ADMIN_USER_PASS=[^{].*/ADMIN_USER_PASS=********/g' ${domainPropertiesOutput}
```

Comment:
Can this now have the RCU pass too?  Maybe just filter out any line with "PASS" (ditto for similar debugging in other scripts).   `grep -iv ".*pass.*=.*" file`

Reply:
Ditto for any similar debugging in other scripts.